### PR TITLE
Count partial mixins as tested deps

### PR DESCRIPTION
**Pull request by lukebjerring**
Fixes a bug with #13611 where tested partial interfaces weren't being recognized as dependencies for the interfaces that use the mixins.

### CODE REVIEW

File: resources/idlharness.js [hunk 1]
```diff
@@ -306,9 +306,13 @@ IdlArray.prototype.add_dependency_idls = function(raw_idls, options)
                     const inheriting = parsed[dep_type];
                     const inheritor = parsed.name || parsed.target;
                     const deps = [inheriting];
-                    // For A includes B, we can ignore A unless B is being tested.
+                    // For A includes B, we can ignore A, unless B (or some of its
+                    // members) is being tested.
                     if (dep_type !== "includes"
-                        || (inheriting in this.members && !this.members[inheriting].untested)) {
+                        || inheriting in this.members && !this.members[inheriting].untested
+                        || this.partials.find(function(p) {
```
Comment (foolip) [inline resources/idlharness.js]:
```suggestion
                        || this.partials.some(function(p) {
```

**Review by foolip — APPROVED** [review]
LGTM % suggestion, and can you write a test for this?

**Comment by foolip** [comment]
@wpt-pr-bot seems to be confused :/

**Comment by Hexcles** [comment]
I think this change now makes sense to me, but only after you explaining to me in length last week. A bit more comment and a test would be great indeed. That said, I haven't reviewed any other changes in idlharness and it could well be the case that this kind of complexity is totally common in idlharness, given the nature of dependency resolution, in which case it might not be necessary to comment too much.

**Comment by lukebjerring** [comment]
Faaairly sure that my tinkering around with regressions will result in the results for the last commit being available in staging.wpt.fyi shortly..

**Comment by lukebjerring** [comment]
https://staging.wpt.fyi/results/?product=chrome%5Bmaster%5D&product=chrome%40340b29725f&diff
https://staging.wpt.fyi/results/?product=firefox%5Bmaster%5D&product=firefox%40340b29725f&diff

@foolip This clearly demonstrates the need for running the affected tests on master, or https://github.com/web-platform-tests/wpt.fyi/pull/722 and other diffing improvements.

We can be cheeky with a few different queries though;
- [idlharness](https://staging.wpt.fyi/results/?product=chrome%5Bmaster%5D&product=chrome%40340b29725f&diff&q=idlharness)
- [interface](https://staging.wpt.fyi/results/?product=chrome%5Bmaster%5D&product=chrome%40340b29725f&diff&q=interface)

**Comment by lukebjerring** [comment]
There's a [regression in selection](https://staging.wpt.fyi/results/selection/idlharness.window.html?product=chrome%5Bmaster%5D&product=chrome%40340b29725f&diff&q=idlharness)

**Comment by jgraham** [comment]
Can this land?

**Comment by lukebjerring** [comment]
Yeah, I think the removed tests as seen in the diff are expected.

Diff URL for https://github.com/web-platform-tests/wpt/pull/13878/commits/c4f9ad65b265413517f2757dd45de77ec4d3440a is:
https://staging.wpt.fyi/results/?product=chrome%5Bmaster%5D&product=chrome%40c4f9ad65b2&diff&filter=ACU

**Pull request merged**